bin/api-docs/gen-block-lib-list.js

```diff
@@ -0,0 +1,175 @@
+const fs = require( 'fs' );
+const { keys } = require( 'lodash' );
+/**
+ * Path to root project directory.
+ *
+ * @type {string}
+ */
+const ROOT_DIR = path.resolve( __dirname, '../..' );
+
+/**
+ * Path to packages directory.
+ *
+ * @type {string}
+ */
+const BLOCK_LIBRARY_DIR = path.resolve(
+	ROOT_DIR,
+	'packages/block-library/src'
+);
+
+/**
+ * Path to docs file.
+ *
+ * @type {string}
+ */
+const BLOCK_LIBRARY_DOCS_FILE = path.resolve(
+	ROOT_DIR,
+	'docs/reference-guides/core-blocks.md'
+);
+
+/**
+ * Start token for matching string in doc file.
+ *
+ * @type {string}
+ */
+const START_TOKEN = '<!-- START Autogenerated - DO NOT EDIT -->';
+
+/**
+ * Start token for matching string in doc file.
+ *
+ * @type {string}
+ */
+const END_TOKEN = '<!-- END Autogenerated - DO NOT EDIT -->';
+
+/**
+ * Regular expression using tokens for matching in doc file.
+ * Note: `.` does not match new lines, so [^] is used.
+ *
+ * @type {RegExp}
+ */
+const TOKEN_PATTERN = new RegExp( START_TOKEN + '[^]*' + END_TOKEN );
+
+/**
+ * Returns list of keys, filtering out any experimental
+ * and wrapping keys with ~~ to strikeout false values.
+ *
+ * @type {Object} obj
+ * @return {string[]} Array of truthy keys
+ */
+function getTruthyKeys( obj ) {
+	return keys( obj )
+		.filter( ( key ) => ! key.startsWith( '__exp' ) )
+		.map( ( key ) => ( obj[ key ] ? key : `~~${ key }~~` ) );
+}
+
+/**
+ * Process list of object that may contain inner keys.
+ * For example: spacing( margin, padding ).
+ *
+ * @param {Object} obj
+ * @return {string[]} Array of keys (inner keys)
+ */
+function processObjWithInnerKeys( obj ) {
+	const rtn = [];
+
+	const kvs = getTruthyKeys( obj );
+
+	kvs.forEach( ( key ) => {
+		if ( Array.isArray( obj[ key ] ) ) {
+			rtn.push( `${ key } (${ obj[ key ].sort().join( ', ' ) })` );
+		} else if ( typeof obj[ key ] === 'object' ) {
+			const innerKeys = getTruthyKeys( obj[ key ] );
+			rtn.push( `${ key } (${ innerKeys.sort().join( ', ' ) })` );
+		} else {
+			rtn.push( key );
+		}
+	} );
+	return rtn;
+}
+
+/**
+ * Augment supports with additional default props.
+ *
+ * The color support if specified defaults background and text, if
+ * not disabled. So adding { color: 'link' } support also brings along
+ * background and text.
+ *
+ * @param {Object} supports - keys supported by blokc
+ * @return {Object} supports augmented with defaults
+ */
+function augmentSupports( supports ) {
+	if ( 'color' in supports ) {
+		// If backgroud or text is not specified (true or false)
+		// then add it as true.a
+		if ( ! ( 'background' in supports.color ) ) {
+			supports.color.background = true;
+		}
+		if ( ! ( 'text' in supports.color ) ) {
+			supports.color.text = true;
+		}
+	}
+	return supports;
+}
+
+/**
+ * Reads block.json file and returns markdown formatted entry.
+ *
+ * @param {string} filename
+ *
+ * @return {string} markdown
+ */
+function readBlockJSON( filename ) {
+	const data = fs.readFileSync( filename, 'utf8' );
```

Comment:
Nitpick: any particular reason why the JSON file isn't loaded with `require` call that would parse to JSON behind the scenes?

Reply:
TIL. Though I probably knew it at some other point too 😆